src/main/java/org/jahia/modules/csp/taglib/ContentSecurityPolicyNonce.java

```diff
@@ -0,0 +1,15 @@
+package org.jahia.modules.csp.taglib;
+
+import org.jahia.modules.csp.AddContentSecurityPolicy;
+import org.jahia.settings.SettingsBean;
+
+/**
+ * Generates nonce placeholder for inline scipts
+ */
+public class ContentSecurityPolicyNonce {
+
+    public static String noncePlaceholder() {
+        String placeHolderName= SettingsBean.getInstance().getPropertiesFile().getProperty(AddContentSecurityPolicy.CSP_NONCE_PLACEHOLDER_PROP, "XXXXX");
+        return String.format("nonce=\"%s\"", placeHolderName);
```

Comment:
With that we have a problem mentioned in the description of the ticket:

> This JSP tag/function could be placed in inline scripts/styles to conditionally add a nonce, when the content-security-policy module is activated for a site. We should not unconditionally put it there for other sites, because this will reveal the secret nonce placeholder in the HTML source, which is a security risk.

Currently if I test it with two sites, one has the CSP activated, the other not - I see this in the source:
```
<script type="text/javascript" nonce="XXXXX">
            videojs.options.flash.swf = 'http://localhost:8080/jahia/modules/assets/swf/video-js.swf';
        </script>
```

XXXXX is the secret nonce placeholder. If it appears in the source it is no longer secret. So attackers can inject a script and add the nonce with the correct placeholder, so the nonce placeholder will be replaced with the random nonce in the filter and thus legitimate the script and make it runable in the browsers.

Reply:
Ah, ok I see what you meant by that description now. I'll fix it.